servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcUtils.java

```diff
@@ -284,9 +285,10 @@ private static void initResponse(final HttpResponseMetaData response, @Nullable
         final HttpHeaders headers = response.headers();
         headers.set(SERVER, GRPC_USER_AGENT);
         headers.set(CONTENT_TYPE, GRPC_CONTENT_TYPE);
-        if (context != null) {
-            headers.set(GRPC_ACCEPT_ENCODING_KEY,
-                    acceptedEncodingsHeaderValueOrCached(context.supportedMessageCodings()));
+        final CharSequence acceptedEncoding = context == null ? null :
```

Comment:
consider avoiding a `null` conditional if not necessary:
```suggestion
          if (context != null) {
            final CharSequence acceptedEncoding =
                    acceptedEncodingsHeaderValueOrCached(context.supportedMessageCodings());
            if (acceptedEncoding != null) {
                headers.set(GRPC_ACCEPT_ENCODING_KEY, acceptedEncoding);
            }
        }
```